### PR TITLE
Improve psalm type inference, add missing type annotations

### PR DESCRIPTION
**Pull request by gsteel**
|    Q          |   A
|-------------- | ------
| Documentation | yes
| QA            | yes

### Description

Reduces psalm's baseline to almost nothing with 99.7% type coverage

### CODE REVIEW

**Review by Ocramius — APPROVED** [review]
LGTM, thanks @gsteel!

**Review by froschdesign — APPROVED** [review]

File: src/ProblemDetailsResponseFactory.php [hunk 1]
```diff
@@ -296,22 +305,27 @@ protected function getThrowableCode(Throwable $e): int
...
      * Ensure all keys in this associative array are valid XML tag names by replacing invalid
      * characters with an `_`.
+     *
+     * @param array<array-key, mixed> $input
+     * @return array<string, mixed>
      */
     private function cleanKeysForXml(array $input): array
     {
         $return = [];
+        /** @psalm-var mixed $value */
```
Comment (boesing) [inline src/ProblemDetailsResponseFactory.php:328]: @Ocramius This is btw why I disabled `MixedAssignment` in my projects. This is 100% useless and its not properly typable and thus `MixedAssignment` is unnecessary.
There are even discussion in the psalm repository where they talk about removing `MixedAssignment` at all as it is not a real issue.

Comment (boesing) [reply src/ProblemDetailsResponseFactory.php:328]: https://github.com/vimeo/psalm/pull/8776

Comment (Ocramius) [reply src/ProblemDetailsResponseFactory.php:328]: IMO worth investigating/baselining: I have `mixed` stuff so rarely nowadays, that this is okay with baselines.

A `mixed` assignment in one of my own projects is usually a bug or design mistake.

Comment (boesing) [reply src/ProblemDetailsResponseFactory.php:328]: Imho a mixed assignment in projects is almost "standard" as you have some stuff like:

```php
$value = $request->getAttribute('whatever');
```

There is almost no way to properly have types for that unless you put that to a service which extracts attributes to DTOs. Imho thats usually overengineered and thus I am fine with something like:


```php
$value = $request->getAttribute('whatever'); // mixed
Assert::positiveInt($value);
```

Same for console applications providing stuff via STDIN. Youcan extract arguments which are properly typed in symfony/console using `addArgument` while psalm is still not able to understand what the value is when read.

And that is fine. 🤷🏼‍♂️ 
If you do not pass `mixed` through half of your application, thats okayish imho.

Comment (Ocramius) [reply src/ProblemDetailsResponseFactory.php:328]: I only ever pass those into an upcast tool, like `Psl\Type` : `laminas/automatic-releases` is an example of that

**Pull request merged**